src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,35 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirmation extends Popup {
+    constructor({ handleFormSubmit, modalSelector, loadingText }) {
+        super({ modalSelector });
+        this._popupForm = this._modalElement.querySelector(".modal__form");
+        this._handleFormSubmit = handleFormSubmit
+        this._loadingText = loadingText
+        
+    }
+
+    setSubmitAction(action) {
+      this._handleFormSubmit = action;
+    }
+
+    renderLoading(isLoading, submitSave) {
+        if (isLoading) {
+          this._popupForm.querySelector(".modal__save-button").textContent = this._loadingText;
+        } else {
+          this._popupForm.querySelector(".modal__save-button").textContent = submitSave;
+        }
+      }
+
+      close() {
+        super.close();
+        this._popupForm.removeEventListener("submit", this._handleFormSubmit);
+
+      }
+    setEventListeners() {
```

Comment:
It's important to format code correctly, use the indentation hierarchy, and split your code. Please check these sources to study the standard: [https://code.s3.yandex.net/web-developer/landings/design-rules-en/index.html](https://code.s3.yandex.net/web-developer/landings/design-rules-en/index.html), [https://javascript.info/coding-style](https://javascript.info/coding-style).

To make sure your code is formatted correctly, we suggest using Prettier (https://prettier.io/). It can be integrated with any code editor or IDE.

You should fix the formatting in all the files